Uzair-Imtiaz/Code-Book
# Models

### Description
This pull request introduces several models that enhance profile management and project feedback features to the application. The new models include Profile, Skill, Project, Review, and Vote, which collectively contribute to the main functionality of the app, i.e, create user profiles, upload projects, leave your votes and reviews on the projects.

### Testing Instructions
To test the newly added features and ensure their proper functioning, follow the steps below:

Clone the repository:
`git clone git@github.com:Uzair-Imtiaz/Code-Book.git`

Create a virtual environment:
`python3 -m venv myenv`

Activate the venv:
`source myenv/bin/activate`

Install the required modules:
`pip install -r requirements.txt`

Navigate to the project directory:
`cd Code-Book`

Checkout to the models branch:
`git checkout models`

Create a super user so that you can have access to the admin interface of the database.
`python3 manage.py createsuperuser`

    - Provide a username.
    - Email is optional.
    - Provide a password.
    - Confirm password.
 Success message will be displayed if the user is created.

Apply database migrations:
`python3 manage.py migrate`

Start the development server:
`python3 manage.py runserver`

Open a web browser and visit the application at **http://localhost:8000/**.

To see the models, visit **http://localhost:8000/admin** and login with the superuser credentials you created.

### Snapshots
<img width="1237" alt="Screenshot 2023-07-09 at 2 48 22 PM" src="https://github.com/Uzair-Imtiaz/Code-Book/assets/107175205/cb8e2f76-de5b-4870-817f-eafef43798d8">
